File: IT Shop/frontend/src/interface/product.ts
export interface Product{
  ID: number;
CreatedAt: string;
UpdatedAt: string;
DeletedAt: string | null;
ProductName: string;
Description: string;
PricePerPiece: number;
Stock: number;
CategoryID: number;
BrandId: number;
Carts: any;
Pictures: any;
}

File: IT Shop/backend/entity/product.go
package entity

import (
	

	"gorm.io/gorm"
)

type Product struct {
    
    gorm.Model
	ProductName    string
	Description     string
	PricePerPiece float64
	Stock           uint

	// fk
	CategoryID uint

	BrandId uint

	// ตัวเชื่อมกับ cart
	Carts []Cart `gorm:"foreignKey:ProductId"`
	Pictures []Picture `gorm:"foreignKey:ProductId"`
}
